### DISCUSSION TITLE
[Menu Entry Swapper] No documentation on how to use customizable left-click on NPCs/Objects?

### DISCUSSION THREAD

**Discussion by christophercarney** [runelite/runelite#15073] [Q&A] [answered]
I'm struggling to find out how to use the menu entry swapper for NPCs and Object Swaps. The current documentation seems out-of-date here: https://github.com/runelite/runelite/wiki/Menu-Entry-Swapper as it only mentions how to configure Item Swaps. How do you configure this feature?

**Comment by Nightfirecat** [comment, marked as answer]
I've updated the wiki page with the following section for those:

* https://github.com/runelite/runelite/wiki/Menu-Entry-Swapper#customizable-inventory-item-swaps
* https://github.com/runelite/runelite/wiki/Menu-Entry-Swapper#customizable-object-swaps

**Reply by christophercarney** [reply]
Thanks! One follow-up, what is the logic behind what objects can be swapped vs. not, I think the plugin text is worded like it allows for swapping of arbitrary objects -- but I can't swap the left-click of a tree from "Chop down" to "Examine" for example

Specifically in my house during build mode I'd like to change the left-click default of certain objects to "Remove"

**Reply by Nightfirecat** [reply]
Examine works differently than other menu options (it doesn't really cause an action) so it's not available to be swapped to.

Construction menu swapping [is a rejected feature, disallowed by Jagex's statement on unofficial clients](https://github.com/runelite/runelite/wiki/Rejected-or-Rolled-Back-Features#menu-entry-swapping):

> `Build/Remove` on POH hotspots: [see Jagex statement.](https://secure.runescape.com/m=news/a=13/another-message-about-unofficial-clients?oldschool=1) (Try [this method](https://www.youtube.com/watch?v=u9AZWsDfo1I) instead)
